File: Backend/src/main/java/br/com/estoque/dto/UsuarioRequest.java
package br.com.estoque.dto;

import br.com.estoque.model.Role;
import lombok.Data;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
public class UsuarioRequest {
    @NotBlank
    private String nome;

    @Email
    @NotBlank
    private String email;

    @NotBlank
    private String senha;

    @NotBlank
    private String telefone;

    @NotBlank
    private String cpf;

    @NotNull
    private Long roleId;

    private EnderecoDTO endereco;

    // Adiciona o id da empresa associada
    @NotNull
    private Long empresaId;

    @Data
    public static class EnderecoDTO {
        private String cep;
        private String rua;
        private String numero;
        private String bairro;
        private String cidade;
        private String estado;
    }
}

File: FrontEnd/controle-estoque/src/types/UsuarioRequest.ts
export interface UsuarioRequest {
  nome: string
  email: string
  telefone: string
  cpf: string
  empresaId: number
  roleId: number
  ativo: boolean
  endereco: {
    cep: string
    rua: string
    numero: string
    complemento?: string
    bairro: string
    cidade: string
    estado: string
  }
  atualizar: boolean // 🔹 adiciona aqui
}
